runtime/major_gc.c

```diff
@@ -1280,6 +1291,9 @@ static intnat major_collection_slice(intnat howmuch,
     /* To handle the case where multiple domains try to finish the major
       cycle simultaneously, we loop until the current cycle has ended,
       ignoring whether caml_try_run_on_all_domains succeeds. */
+
+    atomic_store(&domain_finished_global_roots, 0);
```

Comment:
Not really sure why we are doing this here. Might it be better to do it inside the `cycle_all_domains_callback`, that way all the code for synchronizing is in the same place?

Reply:
If we do it inside `cycle_all_domains_callback`, is there a possibility that the variable set to 1 by one domain could be reset to 0 by some other domain? I thought there was, hence I did it outside `cycle_all_domains_callback`. If that's not an issue, then it is better to put this inside the function.

Reply:
so within `cycle_all_domains_callback` you have the `caml_global_barrier_is_final` block which gives you one domain to do any setup of the variables you need, knowing that all the other domains are waiting for you.

Reply:
Done.